ssm/ssm2.py

```diff
@@ -429,7 +436,13 @@ def start_connection(self):
             log.info('Will send messages to: %s', self._dest)
 
         if self._listen is not None:
-            self._conn.subscribe(destination=self._listen, ack='auto')
+            # Use the current time to create a subscription id to avoid
+            # any potential id clashes
```

Comment:
Clash with what? Self, or other subs? Will need a more robust method if it could clash with other people.

Reply:
good question, i'm not sure! I'll try to find out

Reply:
from [stomp.io](https://stomp.github.io/stomp-specification-1.2.html#SUBSCRIBE_id_Header):

> SUBSCRIBE id Header
> Since a single connection can have multiple open subscriptions with a server, an id header MUST be included in the frame to uniquely identify the subscription. The id header allows the client and server to relate subsequent MESSAGE or UNSUBSCRIBE frames to the original subscription.
>
>Within the same connection, different subscriptions MUST use different subscription identifiers.

So it sounds like we only have to worry about clashing with our subscriptions, and specifically if we ran multiple subscriptions through a single connection (which we don't do now).